### ISSUE TITLE
WRONG EXECUTE_CQL_SCRIPT PATH when adding Command Line Argument to cassandra migration

### ISSUE THREAD

**Issue by dhiresbudhiraja** [jhipster/generator-jhipster#8944]
##### **Overview of the issue**

 ```
[Wed Dec 12 07:38:08 UTC 2018]: Override for local usage
 [Wed Dec 12 07:38:08 UTC 2018]: Start Cassandra migration tool
 [Wed Dec 12 07:38:08 UTC 2018]: Waiting for Cassandra connection...
 [Wed Dec 12 07:38:08 UTC 2018]: Connected to Cassandra cluster
 [Wed Dec 12 07:38:08 UTC 2018]: Use /cql/create-keyspace.cql script to create the keyspace if necessary
 [Wed Dec 12 07:38:09 UTC 2018]: Execute all non already executed scripts from /cql/additionalscripts/
 /usr/local/bin/autoMigrate: line 40: /usr/local/bin/execute-cql.sh: No such file or directory
```
There is no such file as **/usr/local/bin/execute-cql.sh** error when i pass a command line argument for changing the path of cassandra migrations from **cql/changelog** to a different folder in **cql folder** which is in **src/main/resources/config** folder.

Error comes from file **src/main/docker/cassandra/scripts/autoMigrate.sh** in line number 40.

`EXECUTE_CQL_SCRIPT=$SCRIPT_DIR'/execute-cql.sh'`

because when files gets added to docker container they are being added without **.sh extension**
as in **src/main/docker/cassandra/Cassandra-Migration.Dockerfile** file

```
# script to orchestrate the automatic keyspace creation and apply all migration scripts
ADD cassandra/scripts/autoMigrate.sh /usr/local/bin/autoMigrate
RUN chmod 755 /usr/local/bin/autoMigrate

# script to run any cql script from src/main/resources/config/cql
ADD cassandra/scripts/execute-cql.sh  /usr/local/bin/execute-cql
RUN chmod 755 /usr/local/bin/execute-cql

```
##### **Motivation for or Use Case**
When a user wants a different folder other than **cql/changelog** from which user wants **cql scipts** to get migrated.
There is scenario where we want a different folder in **cql** folder from where we want our **cql scipts** to get migrated.

##### **Reproduce the error**
1.) Generate an app using Jhipster choosing cassandra as database.
2.) Create a folder in **src/main/resources/config/cql** named **additionalscripts**.
3.) Now add a **.cql** file in **additionalscripts** folder containing the command to create a table.
**(name it what you want)**
4.) Now create a new **.yml** file in **src/main/docker/** folder named **additionalmigration.yml**
**(which is basically for doing additional migrations from additionalscripts folder we created)**
5.) Add following lines in additionalmigration.yml :-

```
version: '2'
services:
    ${appname}-cassandra-migration:
            extends:
                file: cassandra-migration.yml
                service: ${appname}-cassandra-migration
            environment:
                - CREATE_KEYSPACE_SCRIPT=create-keyspace.cql
            command: /cql/additionalscripts/

```
where **${appname}** is basically name of your app you provided at time of generating.Must replace it with your appname.
**(you can see cassandra.yml file for help)** **(It is for doing migrations from additionalscripts folder)**
6.) Now Run Cassandra :- 
`sudo docker-compose -f src/main/docker/cassandra.yml up`

7.) When all the migrations of changelog gets done, Run Additional Migration :- 
`sudo docker-compose -f src/main/docker/additionalmigration.yml up`

8.) it will reproduce the error. :+1: 

##### **Suggest a Fix**

1.) Open **src/main/docker/cassandra/scripts/autoMigrate.sh**
2.) **On Line number 40 :-** 
`EXECUTE_CQL_SCRIPT=$SCRIPT_DIR'/execute-cql.sh'`

**Remove **.sh** and make it :-**
`EXECUTE_CQL_SCRIPT=$SCRIPT_DIR'/execute-cql'`

**as it is there in next else block you could see it in that file :-** 
`EXECUTE_CQL_SCRIPT="./usr/local/bin/execute-cql"`
**(and it is without .sh extension)**

##### **JHipster Version(s)**
```
filefactory@0.0.0 /home/letsdoit/filefactory
└─┬ UNMET DEPENDENCY generator-jhipster-nav-element@2.0.2
  └── UNMET DEPENDENCY generator-jhipster@5.1.0 

```

##### **JHipster configuration**
<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.filefactory"
    },
    "jhipsterVersion": "5.1.0",
    "applicationType": "monolith",
    "baseName": "filefactory",
    "packageName": "com.filefactory",
    "packageFolder": "com/filefactory",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": false,
    "websocket": false,
    "databaseType": "cassandra",
    "devDatabaseType": "cassandra",
    "prodDatabaseType": "cassandra",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "replaced-by-jhipster-info",
    "clientFramework": "angularX",
    "useSass": false,
    "clientPackageManager": "yarn",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": false
  }
}
</pre>
</details>

##### **Browsers and Operating System**

-   [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by pascalgrimaud** [comment]
@dhiresbudhiraja : would you like to contribute and PR to fix it ?

As you seem to be a Cassandra user, for your information, there is this ticket: https://github.com/jhipster/generator-jhipster/issues/8939

**Comment by dhiresbudhiraja** [comment]
Yes i would like to contribute but i am unable create a PR on this repository as they don't allow me

**Comment by pascalgrimaud** [comment]
@dhiresbudhiraja : you need to fork the project, then clone your fork locally, work on it, push to your fork. Then, github will propose you to do a pull request

**Comment by clun** [comment]
@dhiresbudhiraja : You already provided so much work, are you working on a PR - would be legit ? Otherwise I will give a try to your procedure and see if it works.

**Issue**: if you provide arguments to override default, there is a typo in `EXECUTE_CQL_SCRIPT` env variable
```
# parse arguments
if [ "$#" -gt 0 ]; then
    log "Override for local usage"
    CQL_FILES_PATH=$1
    CREATE_KEYSPACE_SCRIPT="create-keyspace.cql" # default create-keyspace script
    if [ "$#" -eq 2 ]; then
      CREATE_KEYSPACE_SCRIPT=$2
    fi
    CREATE_KEYSPACE_SCRIPT_FOLDER="$(dirname $CQL_FILES_PATH)"
    SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
    EXECUTE_CQL_SCRIPT=$SCRIPT_DIR'/execute-cql.sh'
else
    CQL_FILES_PATH="/cql/changelog/"
    EXECUTE_CQL_SCRIPT="./usr/local/bin/execute-cql"
    CREATE_KEYSPACE_SCRIPT_FOLDER="/cql"
fi
```

**Comment by clun** [comment]
Bug has been reproduced. Fix proposition is working. Will correct on branch `spring-boot_2.1.0`
```
[Fri Jan  4 12:07:24 UTC 2019]: Override for local usage
[Fri Jan  4 12:07:24 UTC 2019]: Start Cassandra migration tool
[Fri Jan  4 12:07:24 UTC 2019]: Waiting for Cassandra connection...
[Fri Jan  4 12:07:25 UTC 2019]: Connected to Cassandra cluster
[Fri Jan  4 12:07:25 UTC 2019]: Use /cql/create-keyspace.cql script to create the keyspace if necessary
[Fri Jan  4 12:07:26 UTC 2019]: Execute all non already executed scripts from /cql/additionalscripts/
[Fri Jan  4 12:07:26 UTC 2019]: execute: /cql/additionalscripts/00000000000000_create-table-cedrick.cql
[Fri Jan  4 12:07:29 UTC 2019]: /cql/additionalscripts/00000000000000_create-table-cedrick.cql executed with success in 2 seconds
[Fri Jan  4 12:07:29 UTC 2019]: Migration done
```

**Comment by DanielFran** [comment]
Close as https://github.com/jhipster/generator-jhipster/pull/9010 has been merged in Spring-boot 2.1.0 branch!
